x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -332,8 +333,31 @@ export class FleetPlugin
 
     this.telemetryEventsSender.start(plugins.telemetry, core);
 
+    const logger = appContextService.getLogger();
+
     return {
-      fleetSetupCompleted: () => Promise.resolve(),
+      fleetSetupCompleted: async () => {
+        try {
+          logger.info('Beginning fleet setup');
+
+          const { nonFatalErrors } = await setupFleet(
+            new SavedObjectsClient(core.savedObjects.createInternalRepository()),
+            core.elasticsearch.client.asInternalUser
+          );
+
+          if (nonFatalErrors.length > 0) {
+            logger.info('Encountered non fatal errors during Fleet setup');
+            formatNonFatalErrors(nonFatalErrors).forEach((error) =>
```

Comment:
@nchaulet added some logging for `nonFatalErrors` here and re-used the formatting logic we have in our `setupHandler` for returning the errors as predictable JSON objects. Will work on getting an example of the output together.